codegen/attribute_analyzer.go

```diff
@@ -0,0 +1,191 @@
+package codegen
+
+import (
+	"goa.design/goa/expr"
+)
+
+type (
+	// AttributeAnalyzer analyzes an attribute and its sub-attributes.
+	AttributeAnalyzer interface {
+		// IsPointer returns true if the attribute being analyzed is a pointer.
+		IsPointer() bool
+		// IsRequired returns true if the Required property is set to true.
```

Comment:
I don't understand that comment as the Required field is a list of names in the parent attribute? which "Required" property is this referring to?